### PR TITLE
feat: get ioapi legacy entity id from externalID

### PR DESCRIPTION
**Pull request by bfellows37**
New Behavior Introduced:

Adds a function that takes an ExternalIDs array and finds the first valid matching entity in the list and returns its numeric ID. Includes various validations (see tests)

### CODE REVIEW

File: externalIDs.go [hunk 1]
```diff
@@ -94,6 +95,36 @@ func FormatQuattroDisplayID(sourceDbCode string, panelID interface{}) string {
...
+
+		//split the externalID into a 3-part array on the : character
+		externalIDParts := strings.Split(externalID, ":")
+		if len(externalIDParts) != 3 {
+			continue
+		}
+
+		// if entity part doesn't match requested type
+		if externalIDParts[1] != entity {
+			continue
```
Comment (Clairevf) [inline externalIDs.go:114]: is it worth throwing an error here with the issue?

Comment (bfellows37) [reply externalIDs.go:114]: We're expected to see it a lot and just skip over it in the case we don't like it.

File: externalIDs.go [hunk 2]
```diff
@@ -94,6 +95,36 @@ func FormatQuattroDisplayID(sourceDbCode string, panelID interface{}) string {
 	return formatQuattroKey(sourceDbCode, quattroDisplayID, fmt.Sprint(panelID))
 }
 
+func GetLegacyIOEntityNumericID(externalIDs []string, entity string) (*int, bool) {
```
Comment (mglrz) [inline externalIDs.go:98]: part of my thinking with this library is to strictly enforce the consistency with which the external id prefixes are used, and that drives the somewhat hardcoded approach used in this library so far.

so i think my preference would be to add the `io:display` prefix to the consts at that top, make this func private, and have a public func that's called something specific like `ParseIODisplayID`.

Comment (bfellows37) [reply externalIDs.go:98]: gotcha, I'm in, we can extend the library if we need other needle entities

**Review by mglrz — APPROVED** [review]

**Pull request merged**